lib/modules/dosomething/dosomething_taxonomy/dosomething_taxonomy.module

```diff
@@ -110,15 +110,21 @@ function dosomething_taxonomy_preprocess_taxonomy_term(&$vars) {
 
 /**
  * Returns array of node nid's with the given Term $tid.
- * @deprecated Use taxonomy_select_nodes instead.
  *
  * @param int $tid
  *   A taxonomy term tid.
  *
  * @return array
  */
 function dosomething_taxonomy_get_campaigns($tid) {
-  return taxonomy_select_nodes($tid, FALSE);
+  $nids = taxonomy_select_nodes($tid, FALSE);
+  $campaigns = [];
+  foreach ($nids as $delta => $nid) {
+    if (node_load($nid)->type === 'campaign') {
```

Comment:
This will make a ton of database queries since it needs to first run the query for `taxonomy_select_nodes` and then independently load each node from the DB. A quick improvement would be to use [`node_load_multiple`](https://api.drupal.org/api/drupal/modules%21node%21node.module/function/node_load_multiple/7.x).